fastestimator/op/tensorop/model/update.py

```diff
@@ -32,35 +35,73 @@ class UpdateOp(TensorOp):
 
     Args:
         model: Model instance compiled by fe.build.
-        loss_name: The name of loss.
+        loss_name: The input loss key.
+        gradients: An optional key containing model gradients. These will be directly applied to the model weights
+            during an update. If not provided, gradients will be computed based on the specified loss_name, which will
+            automatically handle any desired mixed-precision scaling. This argument shouldn't be used if mixed-precision
+            training is enabled.
         mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
             regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
             like "!infer" or "!train".
+        merge_grad: The gradient accumulation times before model update. Ex: if `merge_grad` = 3, for every three Op
+            calls only the third one updates the model. The first two calls only accumulate its gradients. This default
+            value is 1 and it will update the model at every step.
         defer: Whether to defer the actual application of the update until the end of the step. This can be necessary
             in PyTorch when trying to update multiple models which depend on one another (ex. certain GANs). By default,
             all UpdateOps which appear contiguously as the last ops of a Network will be deferred. We hope that you will
             never need to worry about this flag, but it's here for you if you need it.
+
+    Raise:
+        ValueError: When model is mixed-precision and `gradients` is provided.
+        ValueError: Network framework is not one of "tf" or "torch".
+        RuntimeError: If attempting to modify a PyTorch model which relied on gradients within a different PyTorch model
+            which has in turn already undergone a non-deferred update.
     """
     def __init__(self,
                  model: Union[tf.keras.Model, torch.nn.Module],
                  loss_name: str,
                  gradients: Optional[str] = None,
                  mode: Union[None, str, Iterable[str]] = "train",
+                 merge_grad: int = 1,
                  defer: bool = False):
+        self.extra_loss = isinstance(model, tf.keras.Model) and model.losses
+        if gradients is None:
+            super().__init__(inputs=loss_name, outputs=None, mode=mode)
+        elif model.mixed_precision:
+            raise ValueError("Mixed precision training cannot take input gradients, because the gradients need to be "
+                             "computed in this module")
+        else:
+            if self.extra_loss:
```

Comment:
why an else here? now if they are using mixed precision and special loss they won't get both warnings right?

Reply:
fixed